cloudify_hostpool/rest/backend/acquire.py

```diff
@@ -0,0 +1,64 @@
+########
+# Copyright (c) 2015 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# * See the License for the specific language governing permissions and
+# * limitations under the License.
+
+
+import uuid
+
+from cloudify_hostpool.backend import scan
+
+
+def acquire(db):
+    '''"Creates" (acquires) a host.
+
+    :return: host dict if successful, None otherwise.
+    :param db: host database.
+    '''
+    for host in _aquisition_gen(db):
+        acquire = False
+        try:
+            if not db.reserve_host(host['global_id']):
+                continue
+            host = _check_if_alive(db, host)
+            if host['alive']:
+                acquire = True
+                break
+        finally:
+            if not acquire:
+                db.update_host(host['global_id'], {'reserved': False})
+    # No host acquired
+    else:
+        return None
+    host = db.update_host(host['global_id'],
+                          {'host': str(uuid.uuid4()),
+                           'reserved': False})
+    return host
+
+
+def _aquisition_gen(db):
+    for alive in True, False:
+        for host in db.get_host(reserved=False, alive=alive):
+            if host.get('host') is None:
```

Comment:
Yes, indeed. This is already changed in branch "Filters" so you can safely merge this branch and "Filters" together.